src/main/scala/intellij/haskell/psi/HaskellLayoutLexer.scala

```diff
@@ -36,17 +36,18 @@ class HaskellLayoutLexer(private val lexer: Lexer,
     override def toString = s"${elementType.toString} ($start, $end)"
 
     val isEOF: Boolean = elementType.isEmpty
-    val isCode: Boolean = elementType.exists(et => !nonCodeTokens.contains(et)) && !isEOF
+    def isCode: Boolean = elementType.exists(et => !nonCodeTokens.contains(et)) && !isEOF
 
-    val isNextLayoutLine: Boolean = isCode && line.columnWhereCodeStarts.contains(column)
+    def isNextLayoutLine: Boolean = isCode && line.columnWhereCodeStarts.contains(column)
   }
 
-  private val tokens = new ArrayBuffer[Token](40)
+  private val tokens = new ArrayBuffer[Token](100)
   private var currentTokenIndex = 0
 
   private def currentToken = tokens.lift(currentTokenIndex)
 
   def start(buffer: CharSequence, startOffset: Int, endOffset: Int, initialState: Int) {
+    if (tokens.nonEmpty) return
```

Comment:
Hmm, why this check if tokens are not empty?

Reply:
Oops, please delete it! The lexer was invoked twice during debugging and I didn't want my breakpoints to stop twice.